lib/core/typeinfo.nim

```diff
@@ -301,33 +314,32 @@ proc len*(x: Any): int =
 
 
 proc base*(x: Any): Any =
-  ## Returns base Any (useful for inherited object types).
+  ## Returns the base type of `x` (useful for inherited object types).
   result.rawType = x.rawType.base
   result.value = x.value
 
 
 proc isNil*(x: Any): bool =
-  ## `isNil` for an any `x` that represents a cstring, proc or
+  ## `isNil` for an `x` that represents a cstring, proc or
   ## some pointer type.
   assert x.rawType.kind in {tyCString, tyRef, tyPtr, tyPointer, tyProc}
   result = isNil(cast[ppointer](x.value)[])
 
-const
-  pointerLike = when defined(gcDestructors): {tyCString, tyRef, tyPtr, tyPointer, tyProc}
-                else: {tyString, tyCString, tyRef, tyPtr, tyPointer,
-                            tySequence, tyProc}
+const pointerLike =
+  when defined(gcDestructors): {tyCString, tyRef, tyPtr, tyPointer, tyProc}
+  else: {tyString, tyCString, tyRef, tyPtr, tyPointer, tySequence, tyProc}
 
 proc getPointer*(x: Any): pointer =
   ## Retrieves the pointer value out of `x`. `x` needs to be of kind
   ## `akString`, `akCString`, `akProc`, `akRef`, `akPtr`,
-  ## `akPointer`, `akSequence`.
+  ## `akPointer` or `akSequence`.
```

Comment:
pre-existing but:
* the comment and the assert seem to say 2 different things (assert is for x.kind, comment is for x.rawType.kind)
* `when defined(gcDestructors)` affects pointerLike but doc comment doesn't reflect this
* repeating constraint in doc comment and assert is not DRY

## proposal
via https://github.com/nim-lang/RFCs/issues/270
```nim
proc setPointer*(x: Any, y: pointer) {.requires: x.rawType.kind in pointerLike.} =
  if y != nil and x.rawType.kind != tyPointer:
    genericAssign(x.value, y, x.rawType)
```
(obviously, out of scope for this PR, but just wanted to mention this)